src/test/java/ReaderWriterTest.java

```diff
@@ -0,0 +1,91 @@
+package test;
+
+import IO.DbReader;
+import IO.DbWriter;
+import data.Database;
+import data.DatabaseQuery;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+/* Validate that a .txt file can be properly be stored written and read from database.  */
+public final class ReaderWriterTest {
+  private static String inputFile;
+  private static String outputFile;
+  private static InputStream input;
+  private static OutputStream output;
+
+  private Database database = new Database("test");
+  private DbWriter writer = new DbWriter(database, "test", "writingIn");
+  private DbReader reader = new DbReader(database, "test", "writingIn");
+
+  @Before
+  public void setUp() {}
+
+  @Test
+  public void validateDatabaseName() {
+    System.out.println("validateDatabaseName()");
+    /* Validate that place where data is stored is named test. */
+    System.out.println("test = " + database.getDatabase());
+    Assert.assertEquals(database.getDirectoryName(), "test");
+  }
+
+  @Test
+  public void validateWriterRunId() {
+    System.out.println("validateWriterRunId()");
+    /* Validate that writer's runId is test. */
+    Assert.assertEquals(writer.getRunId(), "test");
+  }
+
+  @Test
+  public void validateReaderWriterDatabaseConnect() throws IOException {
```

Comment:
I'm a little confused reading this test with all of the variables with names starting with `read` or `write` and the nested logic. Can you try to streamline and maybe add some comments on what's happening?